src/lib/uptime_service/uptime_service.ml

```diff
@@ -0,0 +1,326 @@
+              ~proof_level:Full ()
+          in
+          let (module M : Transaction_snark.S) = Option.value_exn m in
+          let best_tip = Transition_frontier.best_tip tf in
+          let best_tip_staged_ledger =
+            Transition_frontier.Breadcrumb.staged_ledger best_tip
+          in
+          match
+            Staged_ledger.all_work_pairs best_tip_staged_ledger
+              ~get_state:(fun state_hash ->
+                match Transition_frontier.find tf state_hash with
+                | None ->
+                    Error
+                      (Error.createf
+                         "Could not find state_hash %s in transition frontier \
+                          for uptime service"
+                         (State_hash.to_base58_check state_hash))
+                | Some breadcrumb ->
+                    Ok
+                      (Transition_frontier.Breadcrumb.protocol_state breadcrumb)
+            )
+          with
+          | Error e ->
+              [%log error]
+                "Could not get SNARK work from best tip staged ledger for \
+                 uptime service"
+                ~metadata:[("error", Error_json.error_to_yojson e)] ;
+              return ()
+          | Ok [] ->
+              [%log info] "No SNARK jobs available for uptime service" ;
+              return ()
+          | Ok jobs -> (
+              let last_job = List.last_exn jobs in
+              match%bind
+                One_or_two.Deferred_result.map last_job ~f:(fun single_spec ->
+                    Prod.perform_single worker_state ~message single_spec )
+              with
+              | Error e ->
+                  [%log error]
+                    "Error when computing SNARK work for uptime service"
+                    ~metadata:[("error", Error_json.error_to_yojson e)] ;
+                  return ()
+              | Ok proofs -> (
+                  let proof_len =
+                    bin_write_proof_data proof_buf ~pos:0 proofs
+                  in
+                  let proof_string =
+                    String.init proof_len ~f:(fun ndx -> proof_buf.{ndx})
+                  in
+                  match Base64.encode proof_string with
+                  | Error (`Msg err) ->
+                      [%log error]
+                        "Could not Base64-encode SNARK work for uptime service"
+                        ~metadata:[("error", `String err)] ;
+                      return ()
+                  | Ok snark_work_base64 ->
+                      let state_hash =
+                        Transition_frontier.Breadcrumb.state_hash best_tip
+                      in
+                      let block_data =
+                        { block= block_base64
+                        ; created_at= get_rfc3339_time ()
+                        ; peer_id
+                        ; snark_work= Some snark_work_base64 }
+                      in
+                      send_uptime_data ~logger ~submitter_keypair ~url
```

Comment:
iirc we talked about sending this snark work to the pool as well. Is that tabled for next release?

Reply:
shouldn't be hard to add ... do you think we should do this now?

Reply:
We should if there's time since this is something we'll have to add, sooner or later. 
If we are doing this, we should probably use the `snark-worker-fee` if it is configured and then we'll need to include the fee in `proof_data` for verification.